src/silvimetric/storage.py

```diff
@@ -60,78 +54,62 @@ def create(atts:list[str], resolution: float, bounds: list[float],
         if not ctx:
             ctx = tiledb.default_ctx()
 
-        if len(bounds) == 4:
-            minx, miny, maxx, maxy = bounds
-        elif len(bounds) == 6:
-            minx, miny, minz, maxx, maxy, maxz = bounds
-        else:
-            raise Exception(f"Bounding box must have either 4 or 6 entities. {bounds}")
-
-
-        dims = { d['name']: d['dtype'] for d in pdal.dimensions if d['name'] in atts }
-        xi = floor((maxx - minx) / float(resolution))
-        yi = floor((maxy - miny) / float(resolution))
+        dims = { d['name']: d['dtype'] for d in pdal.dimensions if d['name'] in config.attrs }
+        xi = floor((config.bounds.maxx - config.bounds.minx) / float(config.resolution))
+        yi = floor((config.bounds.maxy - config.bounds.miny) / float(config.resolution))
 
         dim_row = tiledb.Dim(name="X", domain=(0,xi), dtype=np.float64)
         dim_col = tiledb.Dim(name="Y", domain=(0,yi), dtype=np.float64)
         domain = tiledb.Domain(dim_row, dim_col)
 
         count_att = tiledb.Attr(name="count", dtype=np.int32)
         tdb_atts = [tiledb.Attr(name=name, dtype=dims[name], var=True, fill=0)
-                    for name in atts]
+                    for name in config.attrs]
 
         schema = tiledb.ArraySchema(domain=domain, sparse=True,
-            capacity=len(atts) * xi * yi * 10000,
+            capacity=len(config.attrs) * xi * yi * 10000,
             attrs=[count_att, *tdb_atts], allows_duplicates=True)
         schema.check()
 
-        try:
-            proj_crs = pyproj.CRS.from_user_input(crs)
-        except:
-            raise Exception(f"Invalid CRS ingested, {crs}")
-
-        tiledb.SparseArray.create(dirname, schema)
-        with tiledb.SparseArray(dirname, "w") as A:
-            metadata = {'resolution': resolution}
-            metadata['bounds'] = [minx, miny, maxx, maxy]
-            metadata['crs'] = proj_crs.to_string()
+        tiledb.SparseArray.create(config.tdb_dir, schema)
+        with tiledb.SparseArray(config.tdb_dir, "w") as A:
+            metadata = {'resolution': config.resolution}
+            metadata['bounds'] = config.bounds.get()
+            metadata['crs'] = config.crs.to_string()
             A.meta.update(metadata)
 
-        s = Storage(dirname)
+        s = Storage(config, ctx)
 
         return s
 
     def consolidate(self, ctx=None):
         # if not ctx:
         #     ctx = self.ctx
-        tiledb.consolidate(self.tdb_dir)
+        tiledb.consolidate(self.config.tdb_dir)
 
-    def saveMetadata(self, metadata: dict) -> None:
+    def saveConfig(self) -> None:
```

Comment:
Should we leave a `saveMetadata` method for when we're adding other metadata attributes to the db? In shatter I'm planning on adding at least `point_count` as a metadata item. Could just add manually with the tiledb write stream, but might be better to handle it in the storage class

Reply:
Sounds good to me 👍🏻